edb/server/main.py

```diff
@@ -719,5 +763,114 @@ def main_dev():
     main()
 
 
+def _coerce_cfg_value(setting: config.Setting, value):
+    if setting.set_of:
+        return frozenset(
+            config.coerce_single_value(setting, v) for v in value
+        )
+    else:
+        return config.coerce_single_value(setting, value)
+
+
+def initialize_static_cfg(
+    cfg_args: List[str], is_remote_cluster: bool
+) -> Tuple[List[Dict[str, Any]], Dict[str, str]]:
+    result = []
+    backend_settings = {}
+    command_line_argument = "A"
+    environment_variable = "E"
+    spec = config.get_settings()
+
+    def add_config(name, value, type_):
+        setting = spec[name]
+        if is_remote_cluster:
+            if setting.backend_setting and setting.requires_restart:
+                if type_ == command_line_argument:
+                    where = "on command line"
+                else:
+                    where = "as an environment variable"
+                raise server.StartupError(
+                    f"Can't set config {name:r} {where} when using "
+                    f"a remote Postgres cluster"
+                )
+        value = _coerce_cfg_value(setting, value)
+        result.append({
+            "name": name,
+            "value": config.value_to_json_value(setting, value),
+            "type": type_,
+        })
+        if setting.backend_setting:
+            if isinstance(value, statypes.ScalarType):
+                value = value.to_backend_str()
+            backend_settings[setting.backend_setting] = str(value)
+
+    env_value: Any
+    setting: config.Setting
+    for env_name, env_value in os.environ.items():
+        cfg_name = env_name.removeprefix("EDGEDB_SERVER_CFG_")
```

Comment:
There's an issue of namespacing the config vars.  If we only allow builtin configuration values to be passed this way, then it's probably fine, but if we generalize it to namespaced configuration variables, then a different scheme is needed, e.g `EDGEDB_SERVER_CONFIG_<module>_<name>`.  The downside is that for builtin vars this will result in `EDGEDB_SERVER_CONFIG_CFG_FOO`, which is a bit awkward looking.

Reply:
Conclusion from the meeting:

1. For env vars, we prefix whatever qualified config names with `EDGEDB_SERVER_CONFIG_`, for example, `EDGEDB_SERVER_CONFIG_cfg::Config.session_idle_timeout=10m`
2. We'll do the same for command-line arguments, like `--config-cfg::Config.session_idle_timeout 10m`
3. To avoid duplicate config names caused by case sensitivity, we may add a unique constraint on the upper/lower-cased config names.
4. To avoid invalid config names caused by spaces or equal signs, we could add another constraint to either forbid those characters, or allow ASCII only.